llvm/test/CodeGen/RISCV/rvv/extractelt-int-rv32.ll

```diff
@@ -8,7 +8,7 @@
 define signext i8 @extractelt_nxv1i8_0(<vscale x 1 x i8> %v) {
 ; CHECK-LABEL: extractelt_nxv1i8_0:
 ; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetivli zero, 1, e8, mf8, ta, ma
+; CHECK-NEXT:    vsetivli zero, 1, e8, m1, ta, ma
```

Comment:
Is there agreement that m1 compared to mf8 is not a regression? Some microarchitectures have same behavior for M1 and fractional LMUL. I don't know whether there is any microarchitectures that can optimize on the fractional LMUL cases.

Reply:
We already increase LMUL for vmv.s.x and vmv.x.s in RISCVInsertVSETVLI, albeit it to avoid a vsetvli. Perhaps that already sets a precedent.

Reply:
The vmv.x.s instruction doesn't really use the VL or the LMUL. It doesn't require the vector register to be naturally aligned to the LMUL. It only uses SEW to know how many bits to extract and where to sign extend in the scalar register. This makes it different than other instructions so it's hard to know for sure.

Reply:
I don't know of any reason to think increasing to m1 here will have any effect on known processors, but there's a lot of uncertainty here.  I will note that defaulting to smallest-legal-fractional-for-sew in insert vsetvli insertion on this instruction probably wouldn't be too painful, so we can likely reverse this easily if needed.  (Luke, might be worth playing with that just to reduce the diff.)

Reply:
If there was another vector instruction in the sequence that needed some different LMUL other than m1, then that LMUL would have been used instead of m1, since vmv.s.x and vmv.x.s has no impact on LMUL anymore. So I am not worried about the possibility of an extra vsetvli instruction being inserted.

I also don't know of any hardware that has differing performance on fractional LMUL compared to m1. For that reason, I think we could probably hold off on defaulting to the smallest-legal-fractional-for-sew in vsetvli insertion until there was a use case that made it worthwhile.

Reply:
I tried out defaulting to mf8 for vmv.s.x and vmv.x.s here https://github.com/lukel97/llvm-project/commit/59bd2221c1970f5c1254d26196c6990978b88751, and then rebasing this branch on top of it here https://github.com/lukel97/llvm-project/commit/1097e21545607d3aaaa00f821127d666f3da9ac2

Doing it this way moves most of the test diffs into that prior commit, but we still see the same missing vsetvli transforms etc. in it